Docs: Improve wording of the FAQ (#1592)
neutrinojs/neutrino: docs/faq.md

your similar projects. Using a preset rather than a boilerplate keeps this process DRY.

Tools like [Create React App](https://github.com/facebook/create-react-app) have

> been fantastic improvements to the tooling ecosystem, but unfortunately, it only works on specific

Reviewer: Unfortunately "it only works" isn't correct grammar, since the beginning of the sentence starts with "Tools like" which is plural.

This should stay as "only work".

```suggestion
been fantastic improvements to the tooling ecosystem, but unfortunately only work on specific
```

> been fantastic improvements to the tooling ecosystem, but unfortunately, only work on specific

Reviewer: I think this sentence is more correct as it was before?
```suggestion
been fantastic improvements to the tooling ecosystem, but unfortunately only work on specific
```

> environments like React, and do not allow simple extensibility of the build configuration. To 

Reviewer: Trailing whitespace was added here?

```suggestion
environments like React, and do not allow simple extensibility of the build configuration. To
```

Author: While editing it might have got added I didn't observe 😅

answer this, new and similar projects are cropping up to build different types of projects,

> often duplicating efforts that miss out on the best practices to share with the other project

Reviewer: ```suggestion
often duplicating efforts and missing out on opportunities to share best practices with the other project
```

config.plugins.push(new webpack.EnvironmentPlugin(['NODE_ENV']));
```

If you wanted to extend this plugin in your project to add more environment variables, you would most likely
resort to either adding a new instance of the `EnvironmentPlugin` by requiring webpack yourself, or looping through

> webpack's plugin array, removing the plugin, and re-instantiating it with your arguments.

Reviewer: I think the existing wording was better here, as mentioned before:
https://github.com/neutrinojs/neutrino/pull/1592#discussion_r456028828

"own" is referring to two different things, so isn't duplicating, and it's important to the meaning of the sentence. The key part of this paragraph is that how you use code in your "own" project, has different needs from a "shared" project.

Please don't close review comments as "resolved" just because you disagree. That doesn't mean you can't start a discussion to convey your points and try and persuade the reviewer - but shutting down the conversation is a quick way to alienate the project maintainers :-)

Author: I will review the changes once again and will make the necessary changes. And about the part of me marking the comments as 'resolved' , I am sorry for that as I am quite new and is trying to learn the environment of GitHub, thank you for correcting, I will keep that in mind for future instances.

> resort to either adding a new instance of the EnvironmentPlugin by requiring webpack yourself, or looping through
> webpack's plugin array, removing the plugin, and re-instantiating it with your arguments.

Reviewer: These two changes regress the readability IMO. Reverting back:
```suggestion
resort to either adding a new instance of the `EnvironmentPlugin` by requiring webpack yourself, or looping through
webpack's plugin array, removing the plugin, and re-instantiating it with your own arguments.
```

Author: In this, the use of 'own' isn't necessary as its been stated before.

  }
  
  return new webpack.EnvironmentPlugin([...plugin.keys, ...customEnvironmentVariables]);

> }); 

Reviewer: Trailing whitespace was added.

```suggestion
});
```

### Can I just re-use my existing webpack configuration?

> The Neutrino configuration does let you merge a configuration object but does not accept a normal webpack configuration

Reviewer: I prefer this the way it was before :-)

```suggestion
The Neutrino configuration does let you merge a configuration object, but does not accept a normal webpack configuration
```

out of the box. Since Neutrino uses webpack-chain and enforces naming of plugins, rules, and loaders, you must use an
object that corresponds with this "schema". Typically this would involve transforming your webpack configuration to nest
entities requiring a name into an object which maps the name to the entity.

> In short, you must transform your webpack configuration to merge it into the Neutrino configuration cleanly.

Reviewer: This was correct usage before IMO

```suggestion
In short, you must transform your webpack configuration in order to merge it into the Neutrino configuration cleanly.
```

### What is the difference between middleware and presets?

> The term "presets" are a hold-over from Neutrino pre-v5. Presets are middleware, the difference, in reality, is in

Reviewer: The bolding of "are" was intentional here and the previous wording I think was slightly better. Reverting:

```suggestion
The term "presets" are a hold-over from Neutrino pre-v5. Presets **are** middleware, the difference in reality is in
```

object that corresponds with this "schema". Typically this would involve transforming your webpack configuration to nest
entities requiring a name into an object which maps the name to the entity.

> In short, you must transform your webpack configuration to facilitate its merging into the Neutrino configuration cleanly.

Reviewer: I think the existing wording was more accurate here. I mentioned this in the previous review comment, which was marked as resolve with no discussion but the change not made?
https://github.com/neutrinojs/neutrino/pull/1592#discussion_r456029776

Author: Marking review as resolved was quite a mistake, I should have kept open for discussion. I would commit the changes, as now I think prior was better.

### What is the difference between middleware and presets?

> The term "presets" are a hold-over from Neutrino pre-v5. Presets **are** middleware, the difference, in reality, is in

Reviewer: ```suggestion
The term "presets" are a hold-over from Neutrino pre-v5. Presets **are** middleware, the difference in reality, is in
```

### What is the difference between middleware and presets?

> The term "presets" are a hold-over from Neutrino pre-v5. Presets **are** middleware, the difference in reality, is in

Reviewer: ```suggestion
The term "presets" are a hold-over from Neutrino pre-v5. Presets **are** middleware, the difference in reality is in
```
